companion.py

```diff
@@ -189,29 +213,35 @@ def refresh(self):
         s = random.SystemRandom().getrandbits(8 * 32)
         self.state = self.base64URLEncode(s.to_bytes(32, byteorder='big'))
         # Won't work under IE: https://blogs.msdn.microsoft.com/ieinternals/2011/07/13/understanding-protocols/
-        webbrowser.open('%s%s?response_type=code&audience=frontier&scope=capi&client_id=%s&code_challenge=%s&code_challenge_method=S256&state=%s&redirect_uri=%s' % (SERVER_AUTH, URL_AUTH, CLIENT_ID, self.base64URLEncode(hashlib.sha256(self.verifier).digest()), self.state, protocolhandler.redirect))
+        webbrowser.open(
+            '{server_auth}{url_auth}?response_type=code&audience=frontier&scope=capi&client_id={client_id}&code_challenge={challenge}&code_challenge_method=S256&state={state}&redirect_uri={redirect}'.format(  # noqa: E501 # I cant make this any shorter
+                server_auth=SERVER_AUTH,
+                url_auth=URL_AUTH,
+                client_id=CLIENT_ID,
+                challenge=self.base64URLEncode(hashlib.sha256(self.verifier).digest()),
+                state=self.state,
+                redirect=protocolhandler.redirect
+            )
+        )
 
     def authorize(self, payload):
-        # Handle OAuth authorization code callback. Returns access token if successful, otherwise raises CredentialsError
-        if not '?' in payload:
-            print('Auth\tMalformed response "%s"' % payload)
-            raise CredentialsError()	# Not well formed
+        # Handle OAuth authorization code callback.
+        # Returns access token if successful, otherwise raises CredentialsError
+        if '?' not in payload:
+            print('Auth\tMalformed response {!r}'.format(payload))
+            raise CredentialsError('malformed payload')  # Not well formed
 
-        data = urllib.parse.parse_qs(payload[payload.index('?')+1:])
+        data = urllib.parse.parse_qs(payload[payload.index('?') + 1:])
```

Comment:
That `1:` looks weird.  I know what it's doing, but would it hurt to throw `()` around the calculation to make it obvious it's `<this value, plus 1>:` ?

Reply:
can do, though I feel that its clear enough without it, due to order of operations it'll never hit the : first